### DISCUSSION TITLE
Supabase JS, Joins without explicitly defining foreign and primary keys on Schema Level

### DISCUSSION THREAD

**Discussion by ahmedbilal205** [supabase/supabase#26728] [Questions] [answered]
Hi,
I want to Run the following query using SupabaseJS
```
select
  inspection_part.serial_no,
  inspection_part.part_name,
  inspection_part.requested_date_time,
  inspection_part.completed_date_time,
  inspection_part.status,
  inspection_part.notes,
  current_parts_health.parts_health
from
  packages.inspection_part
  join packages.current_parts_health on packages.inspection_part.serial_no = packages.current_parts_health.serial_no;
  ```
  When running this query on supabase SQL editor, it works as expected, it is many to many join. But it is not working in SupabaseJS,
  Here is my SupabaseJS code
  "
    const supaRes = await supabase.from("inspection_part")
    .select(`
    serial_no,
    part_name,
    requested_date_time,
    completed_date_time,
    status,
    notes,
    current_parts_health(part_health)
  `)
  .eq('serial_no', 'current_parts_health.serial_no');
  "
  But I am getting the following error
  "Could not find a relationship between 'inspection_part' and 'current_parts_health' in the schema cache"
looking online it appears they want me to define, primary - foreign key on the tables
But I don't want to define foreign - primary key relation in the tables as it will be many to many join and primary key has to be unique. 

Using Nextjs 14

**Comment by GaryAustin1** [comment, marked as answer]
The REST API requires foreign key relations to do joins. 
You will need to use a view and access that from the REST API.

**Reply by ahmedbilal205** [reply]
Thanks. This should be mentioned somewhere in the documentation.
